internal/test_cases/regex_test_case.go

```diff
@@ -35,6 +35,8 @@ func (t RegexTestCase) Run(shell *shell_executable.ShellExecutable, logger *logg
 		return err
 	}
 
+	t.ExpectedPattern = regexp.MustCompile("^" + t.ExpectedPattern.String())
```

Comment:
@ryan-gang let's just add `^` everywhere instead? It seems weird to have a "RegexTestCase" that doesn't actually behave like a regex. When we move to something like `SingleLineOutputTestCase`, we can automatically add both `^` and `\r\n`.